sound/soc/sof/intel/telemetry.h

```diff
@@ -0,0 +1,42 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause) */
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ * telemetry data in debug windows
+ */
+
+#ifndef _SOF_INTEL_TELEMETRY_H
+#define _SOF_INTEL_TELEMETRY_H
+
+#include "../ipc4-telemetry.h"
+
+#define XTENSA_BLOCK_HDR_VER		2
+#define XTENSA_CORE_DUMP_SEPARATOR	0x0DEC0DEB
```

Comment:
yes, it is a magic number and a marker to make sure the following data is exact core dump. It follows ref FW.

Reply:
Right, so you would need to parse the telemetry slot to find the 'separator' for the core dump section, we just assume that the telemetry is not used for anything else but coredump and it is at the start, which is not the case with the reference firmware?

Reply:
@ujfalusi for current firmware, it is not used for anything else. Later versions of SOF may expand use of the telemetry interface, but the magic value will have to be modified to allow kernel to discover an alternative layout. We added this magic specifically to make this discoverable by the kernel. Reference firmware uses a different layout, but it will anyways not be providing Zephyr-style coredumps, so it's a different beast.

Reply:
@kv2019i, this was my understanding, but we are _not_ discovering it at all and if the layout changes to add other telemetry information than this magic/separator is not going to be in this offset..

A tool used with the reference fw can find this section and it can try to interpret it as what the magic tells it is, but we have different information behind, should the magic/separator be different?

Does the coredump always a fixed size? If so, then I guess it is fine to have it always at the start of the slot and add new (runtime) telemetry information after it?

Reply:
@ujfalusi This is a separate distinct to Zephyr based SOF firmware, and it's on purpose different to the separators used by reference firmware.

If this is later extended, that is a new type of telemetry window content layout, and the FW needs to use a new separator (or via other means, e.g. via a manifest flag or something the driver can read and parse) to allow existing clients (like Linux SOF driver) to distinguish the layout.

For now, there are two known layouts, one is the one used by reference firmware and that we do not supported in SOF driver.

And second is the layout added in this PR, and that is the sole thing SOF driver will support.

This may be later extended, but it has to be done in a way that the driver can adapt to it.